records/labels/dns1123.go

```diff
@@ -0,0 +1,50 @@
+package labels
+
+const DNS1123MaxLength int = 63
+
+// mangle the given name to be compliant as a DNS1123 name "component".
+// the returned result should match the regexp:
+//    ^[a-z0-9]([-a-z0-9]*?[a-z0-9])?$
+// returns "" if the name cannot be mangled.
+func AsDNS1123(name string) string {
+	if name == "" {
+		return ""
+	}
+	sz := len(name)
+	if sz > DNS1123MaxLength {
+		sz = DNS1123MaxLength
+	}
+	last := sz - 1
+	label := make([]byte, sz, sz)
+	ll := 0
+	la := -1 // index of last alphanumeric
+	for _, c := range name {
+		b := dns952table[uint8(c)]
+		switch {
+		case b == -int32('-'):
```

Comment:
I like using int32 because I'm using negative's to represent certain things. A negative rune just seemed weird, or at least more weird than a negative int32.